opentelemetry-exporter-cloud-trace/tests/test_gcp_resource_detector.py

```diff
@@ -46,11 +57,75 @@ def test_finding_gce_resources(self, getter):
         )
 
 
+def pop_environ_key(key):
+    if key in os.environ:
+        os.environ.pop(key)
+
+
+def clear_gke_env_vars():
+    pop_environ_key("CONTAINER_NAME")
+    pop_environ_key("NAMESPACE")
+    pop_environ_key("HOSTNAME")
+
+
+class TestGKEResourceFinder(unittest.TestCase):
+    def tearDown(self) -> None:
+        clear_gke_env_vars()
+
+    @mock.patch("opentelemetry.tools.resource_detector.requests.get")
+    def test_missing_environment_variables(self, getter):
+        getter.return_value.json.return_value = GKE_RESOURCES_JSON_STRING
+
+        # If one of CONTAINER_NAME, NAMESPACE is missing from the environment
+        # return no resources
+
+        pop_environ_key("CONTAINER_NAME")
+        pop_environ_key("NAMESPACE")
+        self.assertEqual(get_gke_resources(), {})
+
+        os.environ["NAMESPACE"] = "namespace"
+        self.assertEqual(get_gke_resources(), {})
+
+        pop_environ_key("NAMESPACE")
+        os.environ["CONTAINER_NAME"] = "container_name"
+        self.assertEqual(get_gke_resources(), {})
+        clear_gke_env_vars()
```

Comment:
this is called in `tearDown`; is it necessary here? (Same comment for other tests)

Reply:
The reason why I do this is do ensure a clean state between tests. Otherwise the order of the tests would matter.